Do not use the bundled jar files for cpptasks and log4j and update the
classpath to lookup under /usr/share/java instead.

Also remove compilers unsupported by debian's ant-contrib-cpptasks package from
the JNI's build definitions and specify a valid soname for the produced
library.
--- a/bindings/java/build.xml
+++ b/bindings/java/build.xml
@@ -21,8 +21,6 @@
   <property name="jni.info.plist" location="build/src/Info.plist"/>
   <property name="jni.info.res" location="build/src/sigar.res"/>
   <property name="jni.libprefix" value=""/> <!-- drop "java" prefix -->
-  <property name="cpptasks.jar"
-            location="hyperic_jni/lib/cpptasks.jar"/>
 
   <condition property="file.version" value="-${sigar.version}">
     <isset property="file.versioned"/>
@@ -49,12 +47,20 @@
   <!-- trick TestFQDN into passing with: ant test -Dsigar.fdqn=foo.bar -->
   <property name="sigar.fqdn" value=""/>
 
+  <path id="debian.classpath">
+    <fileset dir="/usr/share/java/">
+      <include name="log4j-1.2.jar" />
+      <include name="junit.jar" />
+      <include name="ant-contrib-cpptasks.jar" />
+    </fileset>
+  </path>
+
   <path id="libjars">
      <fileset dir="lib" includes="*.jar"/>
   </path>
 
   <path id="alljars">
-     <path refid="libjars"/>
+     <!-- <path refid="debian.classpath"/> -->
      <fileset dir="${sigar-bin}/lib" includes="${sigar.jar}"/>
      <fileset dir="${ant.library.dir}" includes="junit*.jar"/>
   </path>
@@ -99,12 +105,16 @@
            source="1.4" target="1.4"
            sourcepath=""
            debug="true"
+           includeAntRuntime="false"
            classpathref="libjars">
       <src path="${src}"/>
       <src path="${build}/src"/>
       <src path="${jni.src.java}"/>
       <include name="**/*.java"/>
       <exclude name="**/test/*.java" unless="junit.present"/>
+      <classpath>
+         <path refid="debian.classpath" />
+      </classpath>
     </javac>
   </target>
 
@@ -307,6 +317,7 @@
       <manifest>
         <attribute name="Main-Class"
                    value="org.hyperic.sigar.cmd.Runner"/>
+        <attribute name="Class-Path" value="log4j.jar junit.jar" />
       </manifest>
     </jar>
   </target>
@@ -407,9 +418,6 @@
        <include name="lib/*.dylib"/>
        <include name="lib/*.lib"/>
      </fileset>
-     <fileset dir=".">
-       <include name="lib/log4j.jar"/>
-     </fileset>
    </copy>
 
    <!-- header files -->
@@ -512,7 +520,6 @@
        <include name="bindings/java/.classpath"/>
        <include name="bindings/java/.project"/>
        <include name="bindings/java/.sigar_shellrc"/>
-       <include name="bindings/java/lib/log4j.jar"/>
        <include name="bindings/java/lib/junit.jar"/>
        <exclude name="**/.svn/**"/>
        <exclude name="bindings/java/build/**"/>
--- a/bindings/java/hyperic_jni/jni-build.xml
+++ b/bindings/java/hyperic_jni/jni-build.xml
@@ -28,10 +28,10 @@
   <property name="jni.libprefix" value="java"/>
   <property name="jni.libversion" value=""/>
 
-  <property name="cpptasks.jar" location="lib/cpptasks.jar"/>
+  <!-- <property name="cpptasks.jar" location="lib/cpptasks.jar"/>  -->
 
   <taskdef name="cc" classname="net.sf.antcontrib.cpptasks.CCTask">
-    <classpath location="${cpptasks.jar}"/>
+    <classpath location="/usr/share/java/ant-contrib-cpptasks.jar"/>
   </taskdef>
 
   <target name="jni-init">
@@ -269,27 +269,6 @@
         runtime="${cc.runtime}"
 	relentless="false">
 
-      <!-- HPUX -->
-      <compiler name="hp"  if="hpux">
-        <compilerarg value="+Z"/>
-        <compilerarg value="+DD64" if="jni.arch64"/>
-
-        <defineset>
-          <define name="${jni.define.name}_HPUX"/>
-          <define name="_HPUX_SOURCE"/>
-          <define name="_REENTRANT"/>
-        </defineset>
-      </compiler>
-
-      <linker name="hp" if="hpux">
-        <libset if="jni.libset.libs"
-                dir="${jni.libset.dir}"
-                libs="${jni.libset.libs}"/>
-
-        <syslibset libs="nsl"/>
-        <syslibset libs="nm"/>
-      </linker>
-
       <!-- Solaris -->
       <compiler name="gcc" if="solaris">
         <compilerarg value="-O2" if="jni.optim"/>
@@ -341,6 +320,7 @@
                 dir="${jni.dmalloc}/lib"
                 libs="dmallocth"/>
         <linkerarg value="${jni.gccm}" if="jni.gccm"/>
+        <linkerarg value="-Wl,-soname,libsigar.so.1.6"/>
       </linker>
 
       <!-- Darwin -->
@@ -400,6 +380,7 @@
                 dir="${jni.libset.dir}"
                 libs="${jni.libset.libs}"/>
         <syslibset libs="kvm"/>
+        <linkerarg value="-Wl,-soname,libsigar.so.1.6"/>
       </linker>
 
       <!-- OpenBSD -->
@@ -438,24 +419,6 @@
         <syslibset libs="kvm"/>
       </linker>
 
-      <!-- AIX -->
-      <compiler name="xlc_r" if="aix">
-        <compilerarg value="-q64" if="jni.arch64"/>
-        <defineset>
-          <define name="${jni.define.name}_AIX"/>
-          <define name="_ALL_SOURCE"/>
-          <define name="_ANSI_C_SOURCE"/>
-          <define name="_POSIX_SOURCE"/>
-        </defineset>
-      </compiler>
-
-      <linker name="xlc_r" if="aix">
-        <linkerarg value="-b64" if="jni.arch64"/>
-        <libset if="jni.libset.libs"
-                dir="${jni.libset.dir}"
-                libs="${jni.libset.libs}"/>
-      </linker>
-
       <!-- WIN32 -->
       <linker name="msvc" debug="${jni.debug}" if="win32">
         <libset if="jni.libset.libs"
